compiler/rustc_middle/src/ty/codec.rs

```diff
@@ -44,9 +44,9 @@ impl<'tcx, E: TyEncoder<'tcx>> EncodableWithShorthand<'tcx, E> for Ty<'tcx> {
 }
 
 impl<'tcx, E: TyEncoder<'tcx>> EncodableWithShorthand<'tcx, E> for ty::Predicate<'tcx> {
```

Comment:
Yeah, I wasn't sure exactly what it does. I actually thought this might have been the problem when I was having trouble early on, so I removed this impl in another branch. But ultimately I kept it here. Let me know if you want me to remove it.